### PR TITLE
TSLint to ESlint

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**
In this PR I migrated the deprecated TSLint to ESLint for linting our code.
Most of our rules were mapped automatically.
For other issues, I tried to come up with suitable settings.

Then, I fixed all issues for `wave-core-new`, `geoengine-app` and `data-atlas-app`.
Finally, I added linter checks for those three sub-projects to our CI.

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Frontend CI](https://trello.com/c/8jL4QxDp/320-frontend-ci)

**Comment by ChristianBeilschmidt** [comment]
[Lints in CI](https://trello.com/c/QF12SqL0/393-lints-in-ci)

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,121 @@
+{
+    "root": true,
+    "ignorePatterns": ["projects/**/*"],
```
Comment (michaelmattig) [inline .eslintrc.json:3]: why ignore the projects?

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                                "private-field",
+                                "constructor",
+                                "public-method",
+                                "protected-method",
+                                "private-method"
+                            ]
+                        }
+                    }
+                ],
+                "camelcase": "off",
```
Comment (michaelmattig) [inline .eslintrc.json:57]: why turn camelCase off and then enable it again for individual features? Why don't we use default naming-conventions?

File: projects/data-atlas-app/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+    "extends": "../../.eslintrc.json",
+    "ignorePatterns": ["!**/*"],
```
Comment (michaelmattig) [inline projects/data-atlas-app/.eslintrc.json:3]: Can you please explain what this means and why we need it

File: projects/data-atlas-app/src/app/app.component.html [hunk 4]
```diff
@@ -30,11 +30,17 @@
...
+        id="analysisButton"
+        class="mat-elevation-z4"
+        *ngIf="(analysisVisible$ | async) === false"
+        (click)="showAnalysis()"
+    >
         <mat-icon svgIcon="cogs"></mat-icon>
         Analysis
     </button>
-    <mat-card id="analysisCard" *ngIf="analysisVisible$ | async"> Analysis </mat-card>
+    <mat-card id="analysisCard" *ngIf="analysisVisible$ | async"> Analysis</mat-card>
```
Comment (michaelmattig) [inline projects/data-atlas-app/src/app/app.component.html:43]: so trailing spaces at the end is ok but not in front?

**Review by michaelmattig — COMMENTED** [review]
I only skimmed over it. I guess we'll see if this works for us or not in the future. Please answer my questions then can merge I guess.

File: projects/data-atlas-app/src/app/app.component.html (hunk 4 above)
Comment (ChristianBeilschmidt) [reply projects/data-atlas-app/src/app/app.component.html:43]: I'm not sure about this one but spaces in HTML have meanings and you can erase some (e.g. multiple consecutive spaces) and some you can't without changing the behavior :shrug:.

File: .eslintrc.json (hunk 2 above)
Comment (ChristianBeilschmidt) [reply .eslintrc.json:57]: Well, I guess we want to use constants like GENERIC_FIELDS and we can't do this with camelCase. Furthermore, our types from the backend often use snake_case and for enums we usually use PascalCase. So camelCase for everything doesn't work out. But if you have alternative suggestions, feel free to suggest some.

File: .eslintrc.json (hunk 1 above)
Comment (ChristianBeilschmidt) [reply .eslintrc.json:3]: I think it ignores the projects in the root and then has one file for each project that extends this one.

It was generated automatically, so I don't have full insight, but it checks every project in the end.

File: projects/data-atlas-app/.eslintrc.json (hunk 3 above)
Comment (ChristianBeilschmidt) [reply projects/data-atlas-app/.eslintrc.json:3]: I can't tell you unfortunately, it was generated automatically. We could try out different modifications if you like?

**Comment by michaelmattig** [comment]
Could you maybe add a command to package json that does the same as the CI, s.t. we don't need to run 4 separate commands before commit/push?

**Comment by ChristianBeilschmidt** [comment]
> Could you maybe add a command to package json that does the same as the CI, s.t. we don't need to run 4 separate commands before commit/push?


fb9498b

**Pull request merged**